File: curso_c_cpp/cpp/05_constructores_destructores/ejemplos/contador_static.cpp
/*
 * Miembro static: contar cuantos objetos se han creado.
 */
#include <iostream>
using namespace std;

class Usuario {
public:
    static int total;       // compartido por toda la clase
    Usuario() { total++; }
};

int Usuario::total = 0;     // definicion (una sola vez, fuera de la clase)

int main() {
    Usuario a, b, c;
    cout << "Usuarios creados: " << Usuario::total << endl;   // 3
    return 0;
}

File: curso_java/11_constructores_static/ejemplos/ContadorStatic.java
/*
 * Miembro static compartido: contar cuantos objetos se crean.
 */
public class ContadorStatic {
    static int total = 0;        // compartido por toda la clase

    public ContadorStatic() {
        total++;
    }

    public static void main(String[] args) {
        new ContadorStatic();
        new ContadorStatic();
        new ContadorStatic();
        System.out.println("Objetos creados: " + ContadorStatic.total);   // 3
    }
}
